src/commands/deploy/deploy.ts

```diff
@@ -118,3 +119,27 @@ export async function deploy(actionContext: IActionContext, arg1?: vscode.Uri |
     // tslint:disable-next-line: no-floating-promises
     runPostDeployTask(node, correlationId, tokenSource);
 }
+
+async function runDeployTask(context: IDeployContext, client: appservice.SiteClient, siteConfig: WebSiteModels.SiteConfigResource, deployPath: string): Promise<void> {
```

Comment:
"task" has a special VS Code meaning, which doesn't apply here. We could find a new name, but seems easier to just move this code up into line 110 above. You already encapsulated most of the survey stuff into `shouldPromptForSurvey` and `failureMoreInfoSurvey`

Reply:
First time I tried to just keep it in the same function I hit a linting limit on complexity, but it seems to fit under the limit now.